File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>

int count_letters(string text);
int main(void)
{
    string text = get_string("Text: ");

    int letters = count_letters(text);
    int len_text = strlen(text);

    int words = 1;
    for(int i = 0; i < len_text; i++)
    {
        if (text[i] == 32)
        {
            words++;
        }
    }



    int sentences = 0;
    for(int j = 0; j < len_text; j++)
    {
        if(text[j] == 46 || text[j] == 33 || text[j] == 63)
        {
            sentences++;
        }
    }

    int L = letters/words * 100;
    int S = sentences/words * 100;
    int index = 0.0588 * L - 0.296 * S - 15.8;
    printf("Grade %i\n", index);


    printf("%i\n", letters);
    printf("%i\n", words);
    printf("%i\n", sentences);

}




int count_letters(string text)
{
    int lenght = strlen(text);
    int puntuation = 0;
    for(int i = 0; i < lenght; i++)
    {

        if ( text[i] < 97)
        {
            puntuation++;
        }
    }
    int letters = lenght - puntuation;
    return letters;

}

File: sentimental-readability/readability.py
import cs50


def main():
    text = cs50.get_string("text: ")
    letters = count_letters(text)
    len_text = len(text)

    sentences= 0
    words = 1
    for letter in range(len_text):
        if letter == 32:
            words += 1
        elif letter == 46 or letter == 33 or letter == 63:
            sentences +=1

    wor = sentences / words
    l = letters / words * 100
    s = wor * 100
    index = 0.0588 * l - 0.296 * s - 15.8
    grade = index + 0.5

    if grade > 16:
        print("grade 16+")
    elif grade < 1:
        print("before grade 1")
    else:
        print(f"grade {grade}")






def count_letters(text):
    length = len(text)
    puntuation = 0
    for let in range(length):
        let = let.lower()
        if let < 97:
            puntuation += 1
    let = length - puntuation
    return let

main()